src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -846,6 +859,35 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_Vector512_AsVector:
+        case NI_Vector512_AsVector512:
+        {
+            assert(sig->numArgs == 1);
+
+            if (getSIMDVectorRegisterByteLength() == YMM_REGSIZE_BYTES)
+            {
+                assert(IsBaselineVector512IsaSupported());
+                // We support Vector512 but Vector<T> is only 32-bytes, so we should
+                // treat this method as a call to Vector512.GetLower or Vector256.ToVector512
+
+                if (intrinsic == NI_Vector512_AsVector)
+                {
+                    return impSpecialIntrinsic(NI_Vector512_GetLower, clsHnd, method, sig, simdBaseJitType, retType,
+                                               simdSize);
+                }
+                else
+                {
+                    assert(intrinsic == NI_Vector512_AsVector512);
+                    return impSpecialIntrinsic(NI_Vector256_ToVector512, clsHnd, method, sig, simdBaseJitType, retType,
+                                               32);
+                }
+                break;
+            }
+
+            assert(getSIMDVectorRegisterByteLength() == XMM_REGSIZE_BYTES);
```

Comment:
Question - For us to get here  `Vector512.AsVector()` or `Vector512.AsVector512()` has to be lowered which will happen only when target supports `AVX512`(and by extension `AVX2`). Considering this, will we ever end up with a case where `Vector<T>` size = 128 and` Vector512` gets lowered?

Same goes for if this part is necessary? https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/hwintrinsicxarch.cpp#L826-L841


Right now, I'm just letting us fall to software fallback if we end up here

Reply:
It is "dead code" today. We might be able to hit it in release in the future if we end up adding a configuration knob to control the size of `Vector<T>`.

I responded below with the scenario that lets it be hit for `Vector256<T>` vs `Vector<T>`.

Reply:
With `DOTNET_EnableAVX2=0`, I'm able to hit this code for Vector512 too to test it out. In this PR, I'm just not lowering it and keeping the call as is. 
Alternate solutions
1. Treat it as 128 <-> 256 conversion. This seem to work. It'll be a xmm mov so I imagine there are no issues but feel free to correct me 
![image](https://user-images.githubusercontent.com/17969209/227658818-8b1b4dc6-c999-4598-8c82-d084783bc12f.png)

2. Write new `NI_Vector512_GetLower128 `and `NI_Vector128_ToVector512 `atleast internally to handle this. Haven't tried this but I'm open to it

3. Insert 2 nodes -> `NI_Vector512_GetLower `+ `NI_Vector256_GetLower `for eg. Haven't tried this either
Any preference or comments?

Reply:
I think 2 is the "best" long term solution. We'd then be able to recognize 3 in morph and transform it into 2 as an optimization in a follow up PR.

I don't think we can do 1 because the node type/sizes will be mismatched as to the real expectation and that may cause other issues.

I am fine with doing 3 "now" if that is simpler, provided we log an issue tracking the work to enable 2.

Reply:
I decided to go with option 2 too. Added NI_Vector512_GetLower128 and NI_Vector128_ToVector512  to internally handle this. Have pushed this changed